ondine/jse-win15-1.md

```diff
@@ -0,0 +1,216 @@
+### Ondine Gallatin
+
+### Homework #1
+(Due Monday 1/12)
+
+---
+
+**1)** (_Difficulty: mixed_)
+Predict the output of each of the following expressions, and explain the rules which determine it.
+If the output depends on the value of x, identify the conditions when the expression will output true.  Assume the cases are independent, and x is reset to an unknown value before each.
+
+Some of these are tricky!  Don't trust your first instinct.  
+
+
+**a)** `"1" == 1`
+
+True. The string gets converted to a number and 1 == 1.
+
+**b)** `"1" === 1`
+
+Always false. The operator is strict equals - the operands must be of the same data type as well as value.
+
+**c)** `x == 'x'`
+
+False. Any x value does not equal x that is a string. True only if x is assigned the value 'x'. (x = 'x')
+
+**d)** `x == (x+'')`
+
+Always true. The value of x is the same as x plus an empty string.
+
+**e)** `'' == ' '`
+
+Always false. An empty string is not the same as a string that contains a spaceband.
+
+**f)** `x = true`
+
+True. Boolean value.
+
+**g)** `var x; x == 'undefined'`
+
+False. The value of x is not been declared, so its value is undefined. The type of x is "undefined".
+
+**h)** `'9'<'10'`
+
+False. Comparing two strings in order from left to right, 9 !< 1.
+
+**i)** `typeof x + 1 === "number"`
+
+Always false. Whatever x is, the result is not a number. If typeof x is "number", still false because without parentheses, typeof x + 1 returns a string "number1".
+
+**j)** `typeof x % 2 === "number"`
+
+Always false. Whatever x is, the result is not a number. If typeof x is "number", still false
+
+**k)** `typeof (x % 2) === "number"`
+
+Always true. Whatever x is, the type of the remainder is a number.
+
+**l)** `x++ == ++x`
+
+Always false. x++ evaluates to unincremented value of x. ++x evaluates to the incremented value of x.
+
+**m)** `++x == x++`
+
+False if x is not a number value. True if x is a number, because the incremented value of x on the left equals the unincremented value on the right.
+
+**n)** `"1"+x == 1+x`
+
+Only true if x is a string. With any other value, type coercion returns a concatenated string "1value".
+
+**o)** `"0"+1 == 1`
+
+True. Type conversion. "0" converts to 0, then addition makes the result true.
+
+Or - "0" as a string is truthy, numbers other than 0 are truthy.
```

Comment:
You're right, but some minor clarification: truthiness doesn't enter in--autoconversion converts the `1` to a `"1"`, so you get `"01" == 1`. When this is evaluated, JavaScript helpfully assumes we're comparing a couple of numbers, which is why this is true.